Services/src/CSClassroom/CSClassroom.WebApp/Controllers/SubmissionController.cs

```diff
@@ -232,16 +238,124 @@ public async Task<IActionResult> Submit(int userId, int commitId)
 		}
 
 		/// <summary>
-		/// Downloads all submissions for the given section.
+		/// Renders initial entry to the Download Submissions form.  Asks the user to select
+		/// the download format--the components to include in the archive--along with which
+		/// sections and students to download them for.
 		/// </summary>
 		[Route("Submissions/{sectionName}/Download")]
 		[ClassroomAuthorization(ClassroomRole.Admin)]
 		public async Task<IActionResult> Download(string sectionName)
 		{
-			var section = Classroom.Sections
-				.SingleOrDefault(s => s.Name == sectionName);
+			// Gets list of sections and students from db for user to choose from
+			var downloadCandidates = await SubmissionService.GetCheckpointDownloadCandidateListAsync(
+				ClassroomName,
+				ProjectName,
+				CheckpointName);
 
-			if (section == null)
+			List<SectionsAndStudents> sectionStudents = downloadCandidates.Select
+				(
+					dc => new SectionsAndStudents()
+					{
+						SectionId = dc.Section.Id,
+						SectionName = new SelectListItem()
+						{
+							Text = dc.Section.DisplayName,
+							Value = dc.Section.Name,
+
+							// Whichever section the user had clicked on before
+							// choosing to download should be checked by default
+							Selected = (dc.Section.Name == sectionName)
+						},
+						SelectedStudents = dc.Users.Select
+						(
+							user => new StudentToDownload()
+							{
+								Selected = true,
+								FirstName = user.User.FirstName,
+								LastName = user.User.LastName,
+								Id = user.User.Id,
+								Submitted = user.Submitted
+							}
+						).ToList()
```

Comment:
order by last name, then first name? right now order on the page is random

Reply:
Done!  I also added an OrderBy for the section names, as they were random, too.  My test user data unfortunately was already in alphabetical order in the db (when ordered by user id), so I never noticed this.  My test data now has some out-of-order students.